addons/rating/models/mail_thread.py

```diff
@@ -1,12 +1,190 @@
+        record_ids = self.ids
+        result = super(MailThread, self).unlink()
+        self.env['rating.rating'].sudo().search([('res_model', '=', self._name), ('res_id', 'in', record_ids)]).unlink()
+        return result
+
+    def _message_create(self, values_list):
+        """ Force usage of rating-specific methods and API allowing to delegate
+        computation to records. Keep methods optimized and skip rating_ids
+        support to simplify MailThrad main API. """
+        if not isinstance(values_list, (list)):
+            values_list = [values_list]
+        if any(values.get('rating_ids') for values in values_list):
+            raise ValueError(_("Posting a rating should be done using message post API."))
+        return super()._message_create(values_list)
+
+    # RATING CONFIGURATION
+    # --------------------------------------------------
+
+    def _rating_apply_get_default_subtype_id(self):
+        return self.env['ir.model.data']._xmlid_to_res_id("mail.mt_comment")
+
+    def _rating_get_operator(self):
+        """ Return the operator (partner) that is the person who is rated.
+
+        :return record: res.partner singleton
+        """
+        if hasattr(self, 'user_id') and self.user_id.partner_id:
+            return self.user_id.partner_id
+        return self.env['res.partner']
+
+    def _rating_get_parent_field_name(self):
+        """Return the parent relation field name
+           Should return a Many2One"""
+        return None
+
+    def _rating_get_partner(self):
+        """ Return the customer (partner) that performs the rating.
+
+        :return record: res.partner singleton
+        """
+        if hasattr(self, 'partner_id') and self.partner_id:
+            return self.partner_id
+        return self.env['res.partner']
+
+    # RATING SUPPORT
+    # --------------------------------------------------
+
+    def _rating_get_access_token(self, partner=None):
+        """ Return access token linked to existing ratings, or create a new rating
+        that will create the asked token. An explicit call to access rights is
+        performed as sudo is used afterwards as this method could be used from
+        different sources, notably templates. """
+        self.ensure_one()
+        self.check_access_rights('read')
+        self.check_access_rule('read')
+        if not partner:
+            partner = self._rating_get_partner()
+        rated_partner = self._rating_get_operator()
+        ratings = self.rating_ids.sudo().filtered(lambda x: x.partner_id.id == partner.id and not x.consumed)
+        if not ratings:
+            rating = self.env['rating.rating'].sudo().create({
+                'partner_id': partner.id,
+                'rated_partner_id': rated_partner.id,
+                'res_model_id': self.env['ir.model']._get_id(self._name),
+                'res_id': self.id,
+                'is_internal': False,
+            })
+        else:
+            rating = ratings[0]
```

Comment:
It can be simplified a bit (by removing the else clause, and without iterating on all ratings, it will stop after the first one that match the condition)
```suggestion
        rating = next(
            r for r in self.rating_ids.sudo()
            if r.partner_id.id == partner.id and not r.consumed,
            None)
        if not rating:
            rating = self.env['rating.rating'].sudo().create({
                'partner_id': partner.id,
                'rated_partner_id': rated_partner.id,
                'res_model_id': self.env['ir.model']._get_id(self._name),
                'res_id': self.id,
                'is_internal': False,
            })
```

Reply:
Yes, it’s better, thanks